### ISSUE TITLE
[BUG] Opacity changes don't transition with keyframes

### ISSUE THREAD

**Issue by destefanis** [framer/motion#1872]
**2. Describe the bug**

Using opacity with keyframes doesn't work in newer versions of Framer Motion. Only the initial value is rendered.
This may be a newer issue that regressed when the "times" array wasn't working correctly in previous versions. However, I changed the version back to 7.5 and opacity works with keyframes here.

For example:

```jsx
scale: [1, 0.98, 1.02, 1],
y: [0, -20, 0, -2],
opacity: [1, 0.8, 1, 0.8],
transition={{
            duration: 2.5,
            type: "spring",
            bounce: 1,
            times: [0, 0.1, 0.15, 1],
            repeat: Infinity,
            repeatType: "Loop"
          }}
```

In this example, scale and y will animate, and opacity will not.

**3. IMPORTANT: Provide a CodeSandbox reproduction of the bug**

https://codesandbox.io/s/call-animation-7ryx2w?file=/src/App.js:666-880
(change the version to the most recent framer motion version here and you'll see the bug).

A CodeSandbox minimal reproduction will allow us to quickly follow the reproduction steps. **Without one, this bug report won't be accepted.**

**4. Steps to reproduce**

1. Go to framer motion dependency
2. Change to the most recent version
3. Watch the animation and how opacity doesn't change.

**5. Expected behavior**

Opacity should animate with set keyframes.

**Comment by mattgperry** [comment]
I think the bug here is there are multiple keyframes which previously would force the animation the to be of type `"keyframes"` whereas now `type: "spring"` is overriding this.

```
            transition={{
              duration: 2.5,
              times: [0, 0.1, 0.15, 0.18, 0.2, 1],
              repeat: Infinity,
              repeatType: "Loop"
            }}
```

Will look into it, in the meantime the above configuration is working in `8.1.7`. `8.1.8` has a bug with `repeat: Infinity` and WAAPI springs that I'm also looking into.

**Comment by mattgperry** [comment]
That latter bug should be fixed in `8.1.9` and I'll check out why this is running with a spring animation.

**Comment by mattgperry** [comment]
My mistake, this was actually fixed by `8.1.9` too
